packages/color-slider/src/ColorSlider.ts

```diff
@@ -0,0 +1,345 @@
+    public sliderHandlePosition = 0;
+
+    @property({ type: String })
+    public get color(): ColorValue {
+        switch (this._format.format) {
+            case 'rgb':
+                return this._format.isString
+                    ? this._color.toRgbString()
+                    : this._color.toRgb();
+            case 'prgb':
+                return this._format.isString
+                    ? this._color.toPercentageRgbString()
+                    : this._color.toPercentageRgb();
+            case 'hex':
+            case 'hex3':
+            case 'hex4':
+            case 'hex6':
+                return this._format.isString
+                    ? this._color.toHexString()
+                    : this._color.toHex();
+            case 'hex8':
+                return this._format.isString
+                    ? this._color.toHex8String()
+                    : this._color.toHex8();
+            case 'name':
+                return this._color.toName() || this._color.toRgbString();
+            case 'hsl':
+                return this._format.isString
+                    ? this._color.toHslString()
+                    : this._color.toHsl();
+            case 'hsv':
+                return this._format.isString
+                    ? this._color.toHsvString()
+                    : this._color.toHsv();
+            default:
+                return 'No color format applied.';
+        }
+    }
+
+    public set color(color: ColorValue) {
+        if (color === this.color) {
+            return;
+        }
+        const oldValue = this._color;
+        this._color = new TinyColor(color);
+        const format = this._color.format;
+        let isString = typeof color === 'string' || color instanceof String;
+
+        if (format.startsWith('hex')) {
+            isString = (color as string).startsWith('#');
+        }
+
+        this._format = {
+            format,
+            isString,
+        };
+
+        if (isString && format.startsWith('hs')) {
+            const hueExp = /^hs[v|va|l|la]\((\d{1,3})/;
+            const values = hueExp.exec(color as string);
+
+            if (values !== null) {
+                const [, h] = values;
+                this.value = Number(h);
+            }
+        } else if (!isString && format.startsWith('hs')) {
+            const colorInput = this._color.originalInput;
+            const colorValues = Object.values(colorInput);
+            this.value = colorValues[0];
+
+            // The below code line causes some tests to fail
+            //this.value = parseFloat((color as HSV).h.toString());
+        } else {
+            const { h } = this._color.toHsv();
+            this.value = h;
+        }
+        this._previousColor = oldValue;
+        this.requestUpdate('color', oldValue);
+    }
+
+    private _color = new TinyColor({ h: 0, s: 1, v: 1 });
+
+    private _previousColor = new TinyColor({ h: 0, s: 1, v: 1 });
+
+    private _format: { format: string; isString: boolean } = {
```

Comment:
Oh, this one is an object. I prefer this, can we make sure all of the `_format` properties use this shape.